src/ethereum_spec_tools/lint/lints/glacier_forks_hygiene.py

```diff
@@ -0,0 +1,183 @@
+        emit diagnostics.
+        """
+        diagnostics: List[Diagnostic] = []
+
+        all_items = set(previous.keys()) | set(current.keys())
+
+        for item in all_items:
+            try:
+                previous_item = previous[item]
+            except KeyError:
+                self.add_diagnostic(
+                    diagnostics, f"{item} in {name} has been added"
+                )
+                continue
+
+            try:
+                current_item = current[item]
+            except KeyError:
+                self.add_diagnostic(
+                    diagnostics, f"{item} in {name} has been deleted"
+                )
+                continue
+
+            if item == "BOMB_DELAY_BLOCKS":
+                previous_item.value.value = self.delay_blocks[fork_name]
+
+            if not compare_ast(previous_item, current_item):
+                self.add_diagnostic(
+                    diagnostics, f"the item `{item}` in `{name}` has changed"
+                )
+
+        return diagnostics
+
+
+class _Visitor(ast.NodeVisitor):
+    """
+    Visits nodes in a syntax tree and collects functions, classes, and
+    assignments.
+    """
+
+    path: List[str]
+    _items: dict
+
+    def __init__(self) -> None:
+        self.path = []
+        self._items = {}
+
+    def generic_visit(self, node: ast.AST) -> None:
+        """
+        Called if no explicit visitor function exists for a node.
+        Do not visit any child nodes.
+        """
+        sys.exit(f"No visit function defined for {node}")
+
+    def _insert(self, name: str, node: ast.AST) -> None:
+        item = ".".join(self.path + [name])
+        if item in self._items:
+            raise ValueError(f"duplicate path {item}")
+        self._items[item] = node
+
+    @property
+    def items(self) -> Dict:
+        """
+        Sequence of all identifiers found while visiting the source.
+        """
+        return self._items
+
+    def visit_Module(self, module: ast.Module) -> None:
+        """
+        Visit a python module.
+        """
+        for item in module.__dict__["body"]:
+            if type(item) in (ast.Expr, ast.Import, ast.ImportFrom):
+                continue
+            self.visit(item)
+
+    def visit_AsyncFunctionDef(self, function: ast.AsyncFunctionDef) -> None:
+        """
+        Visit an asynchronous function.
+        """
+        self._insert(function.name, function)
+
+    def visit_FunctionDef(self, function: ast.FunctionDef) -> None:
+        """
+        Visit a function.
+        """
+        self._insert(function.name, function)
+
+    def visit_ClassDef(self, klass: ast.ClassDef) -> None:
+        """
+        Visit a class.
+        """
+        self._insert(klass.name, klass)
+
+    def visit_Assign(self, assign: ast.Assign) -> None:
+        """
+        Visit an assignment.
+        """
+        self._insert(assign.targets[0].id, assign)  # type: ignore
```

Comment:
Why do we ignore types here, and is there a way to implement this without ignoring?

Reply:
Done. Explicitly handle when target is name. Notify when it is something else (should ideally not happen in the specs code).